### DISCUSSION TITLE
How to get listitem comment replies

### DISCUSSION THREAD

**Discussion by brunostampfli** [pnp/powershell#4627] [Q&A] [answered]
I have pages with comments and replies to these comments.
I can get the comments with:

`$comments = Get-PnPListItemComment -List "SitePages" -Identity $ItemId`

On a comment I can see the count of replies with:

`$comments[0].ReplyCount`

The result is 1 reply.

How can I get this replay?
With PnP PowerShell?
Or with a CSOM call?

**Comment by waaromikniet** [comment]
I think the only way at this moment is using a Graph query. CSOM and PnP.PowerShell can't get the replies on a comment

**Reply by brunostampfli** [reply]
Hi waaromikniet
I would like to use Graph to get comments and replies.
Do you have any idea which Graph endpoint I can use?

**Reply by waaromikniet** [reply, marked as answer]
I just discovered that there is a SharePoint rest endpoint you can use.

`/_api/web/lists/getByTitle('Site%20Pages')/items(2)/comments(1)/replies`

I can't find a correct Graph endpoint so maybe my assumption was wrong that you could use Graph api

**Reply by brunostampfli** [reply]
Thank you! Works perfekt for me.

```
$url = "$siteurl/_api/web/Lists('0dde4eed-07b9-4a84-bf56-917f06ec025a')/Items(40)/comments(7)/replies"
$output = Invoke-PnPSPRestMethod -Url $url
$output.value
```

```
author        : ...
createdDate   : 2024-12-06T10:24:04.28Z
id            : 8
isLikedByUser : False
isReply       : True
itemId        : 40
likeCount     : 0
listId        : 0dde4eed-07b9-4a84-bf56-917f06ec025a
mentions      : {}
modifiedDate  :`
parentId      : 7
replyCount    : 0
text          : This is an answer.
```
